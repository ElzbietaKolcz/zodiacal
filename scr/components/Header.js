import React, { useState, useEffect } from "react";
import { View } from "react-native";
import { Button, Text } from "react-native-paper";
import tw from "twrnc";
import { auth } from "../../firebase";
import { useDispatch } from "react-redux";
import { logout } from "../features/userSlice";

function Header() {
  const dispatch = useDispatch();
  const [displayName, setDisplayName] = useState(null);

  useEffect(() => {
    const unsubscribe = auth.onAuthStateChanged((user) => {
      if (user) {
        setDisplayName(user.displayName);
      }
    });

    return unsubscribe;
  }, []);

  const handleLogout = async () => {
    try {
      await auth.signOut();
      console.log("User successfully logged out");
      dispatch(logout());
    } catch (error) {
      console.error("Error logging out:", error.message);
    }
  };

  return (
    <View style={tw`w-full flex-row justify-between items-center p-4`}>
      <Text
        variant="headlineSmall"
        style={tw`p-2 mt-1 text-black font-medium`}
        testID="header-welcome-text"
      >
        Welcome {displayName}!
      </Text>
      <View style={tw`flex-row items-center justify-end `}>
        <Button
          style={tw`bg-fuchsia-700  rounded-lg`}
          mode="contained"
          textColor="#FFFFFF"
          onPress={handleLogout}
          testID="button"
        >
          Logout
        </Button>
      </View>
    </View>
  );
}

export default Header;
