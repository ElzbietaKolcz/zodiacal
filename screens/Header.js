import React from 'react';
import { View } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { logout, selectUser } from "../features/userSlice";
import { auth } from "../firebase";
import { Text, Button } from "react-native-paper";
import tw from "twrnc";

function Header({ userName }) {
  const dispatch = useDispatch();
  const logoutOfApp = () => {
    dispatch(logout());
    auth.signOut();
  };

  return (
    <View style={tw`w-full flex-row justify-between items-center p-4`}>
      <Text
        variant="headlineSmall"
        style={tw`p-4 mt-1 text-black font-medium`}
      >
        Welcome back {userName}!
      </Text>

      <View style={tw`flex-row items-center justify-end `}>
        {userName ? (
          <Button
            style={tw`bg-fuchsia-700  rounded-lg`}
            mode="contained"
            textColor="#FFFFFF"
            onPress={logoutOfApp}
          >
            Logout
          </Button>
        ) : null}
      </View>
    </View>
  );
}

export default Header;
