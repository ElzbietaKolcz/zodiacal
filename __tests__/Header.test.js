import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render, fireEvent } from '@testing-library/react-native';
import Header from '../screens/Header';

const mockStore = configureStore([]);

describe('Header component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      user: {
        userName: 'JohnDoe',
      },
    });
  });

  it('renders correctly with user name', () => {
    const tree = render(
      <Provider store={store}>
        <Header />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispatches logout action when logout button is pressed', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Header />
      </Provider>,
    );
    // Log the rendered output for debugging
    console.log(getByTestId('button'));
  
    const logoutButton = getByTestId('button');
    fireEvent.press(logoutButton);
  
    const actions = store.getActions();
    expect(actions).toContainEqual({ type: 'user/logout' });
  });
});
