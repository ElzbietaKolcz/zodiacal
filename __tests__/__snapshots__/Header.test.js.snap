// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly with user name 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 24,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 32,
          },
          {
            "color": "#000",
            "fontWeight": "500",
            "marginTop": 4,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ],
      ]
    }
  >
    Welcome back 
    !
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#a21caf",
          "borderRadius": 8,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#a21caf",
            "borderColor": "transparent",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              Logout
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
